providers/push-api/src/lib/push-api.provider.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  IPushOptions,
+  IPushProvider,
+} from '@novu/stateless';
+import webpush from 'web-push';
+import crypto from 'crypto';
+
+class ErrorBase extends Error {
+  constructor(message?: string) {
+    super(message);
+
+    this.name = this.constructor.name;
+    Error.captureStackTrace(this, this.constructor);
+  }
+}
+
+class PushError extends ErrorBase {
+  constructor(public readonly failures: webpush.SendResult[], message: string) {
+    super();
+    Object.setPrototypeOf(this, new.target.prototype);
+    this.failures = failures;
+    this.message =
+      message +
+      '\nFailures:\n' +
+      failures.map((failure) => failure.statusCode).join('\n');
+  }
+}
+
+export class PushApiPushProvider implements IPushProvider {
+  channelType = ChannelTypeEnum.PUSH as ChannelTypeEnum.PUSH;
+  id: 'push-api';
+
+  constructor(
+    private config: {
+      vapidPrivateKey: string;
+      vapidPublicKey: string;
+    }
+  ) {}
+
+  async sendPush(
+    title: string,
+    content: string,
+    subscription: webpush.PushSubscription
+  ) {
+    const pushOptions: webpush.RequestOptions = {
+      vapidDetails: {
+        subject: 'mailto:support@novu.co',
```

Comment:
What is the purpose of this subject?

Reply:
Vapid subject is to identify the web push server. Technically it should be another credential. I can add this one in there if there isn't already an email credential.

Reply:
@johnmarksilly please go ahead and add this as an email credential.